### PR TITLE
added 0xCB/1337 keyboard

### PR DESCRIPTION
**Pull request by Conor-Burns**
Signed-off-by: Conor Burns <mail@conor-burns.com>

Added the 1337 macropad

## Description

- create keyboard folder 0xcb with model 1337
- added default function for OLED, encoder and via support.
- added some basic functionality in the 4 layers for lighting
- made encoder behave differently for layers
- board will be sold on Tindie as a kit or fully assembled with a 3D-Printed case and the PCB design is open-sourced on my Github

## Types of Changes

- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)

## Issues Fixed or Closed by This PR

*  none :)

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/0xcb/1337/rules.mk [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (drashna) [inline keyboards/0xcb/1337/rules.mk:32]:
```suggestion
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/0xcb/1337/rules.mk [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+ENCODER_ENABLE = yes
+VIA_ENABLE = yes
```
Comment (drashna) [inline keyboards/0xcb/1337/rules.mk:35]: This is not supposed to be enabled at the keyboard level.  

Please create a `via` keymap that has via enabled. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/0xcb/1337/rules.mk (hunk 2 above)
Comment (Conor-Burns) [reply keyboards/0xcb/1337/rules.mk]: Ok resolved - thank you!
make :all works :)

**Review by drashna — APPROVED** [review]

File: keyboards/0xcb/1337/readme.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    
+Compile example:
+
+    qmk compile
+
+Flashing example for this keyboard:
+
+    qmk flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (zvecr) [inline keyboards/0xcb/1337/readme.md:29]:
```suggestion
# 0xCB 1337

Macro keypad

* Keyboard Maintainer: [Conor Burns](https://github.com/conor-burns)
* Hardware Supported: https://github.com/conor-burns/0xcb-1337
* Hardware Availability: Soon on tindie or order your own parts - the hardware in the repo is Open Source :D

To go to bootloader press row 0 col 0 key (top left) while plugging in the board.

Make example for this keyboard (after setting up your build environment):

    make 0xcb/1337:default

Flashing example for this keyboard:

    make 0xcb/1337:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/0xcb/1337/rules.mk [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (zvecr) [inline keyboards/0xcb/1337/rules.mk:11]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/0xcb/1337/readme.md (hunk 3 above)
Comment (Erovia) [reply keyboards/0xcb/1337/readme.md]: I'd also like to ask for a link to an actual picture (or render) of the board.
Preferably with visible PCB.

**Review by Erovia — APPROVED** [review]

File: keyboards/0xcb/1337/1337.c [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+#include "1337.h"
+
+void eeconfig_init_kb(void) {
+#ifdef BACKLIGHT_ENABLE
+    backlight_enable();
+    backlight_level(5);
+#endif
+#ifdef RGBLIGHT_ENABLE
+    rgblight_enable();
+    rgblight_sethsv(0, 255, 255);
```
Comment (fauxpark) [inline keyboards/0xcb/1337/1337.c:26]: Just a note here, you can actually configure these with defines instead now: #11912

**Pull request merged**